### DISCUSSION TITLE
⚡⚡Referral Program : Refer and Earn 💰

### DISCUSSION THREAD

**Discussion by Rana-Biswas** [inspireui/support#5782] [Working on Features]
### Objective: 
A feature that lets users refer the app to another new user through a unique trackable code and earn rewards.

### Plugin Suggestion:
[TeraWallet – Best WooCommerce Wallet System With Cashback Rewards, Partial Payment, Wallet Refunds](https://wordpress.org/plugins/woo-wallet/)
The wallet feature of this plugin is already integrated with Fluxstore; you have to integrate the referral feature only.

![Screenshot 2024-02-05 at 12 43 16 PM](https://github.com/inspireui/support/assets/113324396/c6c8d1ec-ed19-4f87-8158-73564d1aed05)

### Benefits:

1. Both users can earn rewards.
2. Generates more sales.
3. The admin can restrict the rewards to being used in-app only, or the rewards can be withdrawn or transferred to another user as per the admin's requirements.
4. A proper transaction history of all the transactions is already displayed in the Fluxstore Apps at **My wallet** section. 

![Screenshot 2024-02-05 at 2 45 42 PM](https://github.com/inspireui/support/assets/113324396/e0bd0c3c-1379-4284-820e-765b48b7893b)

5. Admin can also set the **Minimum Order Amount** for the referral reward.

![Screenshot 2024-02-05 at 12 51 23 PM](https://github.com/inspireui/support/assets/113324396/da1962cb-b19d-41fd-ae70-d35cc234645a)

### Placement Ideas:

1. A referral screen can be shown after successful sign-up and before showing the homepage, similar to: 

![referral_code](https://github.com/inspireui/support/assets/113324396/65a62a26-12c8-40e5-96b0-94013642b080)

2. It can also be shown in the **Profile Tab** and in the **Side Menu**. 

![Screenshot 2024-02-05 at 2 53 55 PM](https://github.com/inspireui/support/assets/113324396/27352df2-05ef-4504-a6dc-30cced7b07c7)

3. Referrer Screen Idea (it should show the Copy Code button, Total Earnings through referring and Number of Referred users): 

![302240732-f1d43814-b3b2-4d6e-92af-1363cd91d6a5](https://github.com/inspireui/support/assets/113324396/5e594e48-5d0b-4b88-a2e8-0a7d68c66367)

### Enhancements:

1. Fluxbuilder should allow adding a banner that explains how the Referral Program works and what the rewards are.
2. A deep link on the referrer's screen to share the app in one click and it must record this conversion and credit the reward accordingly.

![39790562429dfefc244b5644bf76a1df](https://github.com/inspireui/support/assets/113324396/b3b374dd-59e7-41cf-bf61-b954b091aa80)

**All the e-commerce and multivendor platforms have referral programmes, and I would love to see this on Fluxstore too.**

**Comment by osman-kocak** [comment]
Awesome

**Reply by Rana-Biswas** [reply]
I appreciate your vote. Thanks

**Comment by osman-kocak** [comment]
@minhcasi @HanielQC @luyx2412

**Comment by kientv** [comment]
Is this feature ready?

**Comment by osman-kocak** [comment]
no

Trần Văn Kiên ***@***.***>, 22 Eki 2024 Sal, 14:20 tarihinde
şunu yazdı:

**Comment by Rana-Biswas** [comment]
I paid for this feature and i have it exclusively for my source code

**Reply by osman-kocak** [reply]
Where did you buy it? I want to buy it.

**Reply by Rana-Biswas** [reply]
Paid directly to them InspireUI

**Reply by osman-kocak** [reply]
Can u sell me or how can ı buy on owner. From support ?

**Reply by Rana-Biswas** [reply]
Yes I can, dm me on Instagram and there I will share my WhatsApp number and let's continue there (@ranabiswas9980)

**Reply by osman-kocak** [reply]
Send it

**Reply by kientv** [reply]
@osman-kocak 
Instagram @tranchannhan please

**Reply by osman-kocak** [reply]
Send it

**Comment by cashbackjourney** [comment]
This referral feature sounds really useful and well thought out! I’ve seen similar implementations in apps like **[Zokera](https://zokera.com/)**, where the wallet system combined with referral tracking makes earning rewards seamless for users. Having a unique, trackable referral code definitely helps both the app and users to monitor their earnings clearly. Looking forward to seeing this integrated—it can really boost user engagement and growth.
